### PR TITLE
update macos image from 11 to latest

### PR DESCRIPTION
**Pull request by weshaggard**

### CODE REVIEW

**Review by benbp — APPROVED** [review]

File: eng/pipelines/templates/stages/platform-matrix.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "include": [
     {
       "OSConfiguration": {
-        "macOS-11": {
+        "macos-latest": {
           "OSVmImage": "env:MACVMIMAGE",
           "XCODE_VERSION": "12.5.1"
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/platform-matrix.json:10]:
```suggestion
          "XCODE_VERSION": "14.2"
```

**Comment by weshaggard** [comment]
@LarryOsterman @ahsonkhan any thoughts on the error we are hitting with the MacOS update?
```
/Users/runner/work/1/s/sdk/eventhubs/azure-messaging-eventhubs/test/perf/inc/azure/messaging/eventhubs/test/eventhubs_batch_perf_test.hpp:180:14: error: variable 'total' set but not used [-Werror,-Wunused-but-set-variable]
      size_t total = 0
```

It does appear to be an unused variable but I'm not sure why it isn't triggering on the other OSes.

**Comment by ahsonkhan** [comment]
I believe the strictness of unused variable warnings and flags are compiler specific. So, the compiler on MacOS (clang?) happens to catch it but others like on Windows (MSVC) doesn't.

That's one of the reasons why we test on multiple OSes, to get such coverage.

The change seems reasonable for this PR, especially since it is in tests. That said, I'll let Larry follow-up on whether we should remove the lines all together, since he owns EventHubs.

File: eng/pipelines/templates/stages/platform-matrix-live.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "include": [
     {
       "StaticConfigs": {
-        "macOS-11": {
+        "macos-latest": {
```
Comment (ahsonkhan) [inline eng/pipelines/templates/stages/platform-matrix-live.json:8]: nit: The pipeline name on GitHub CI shows as "macoslatest" which is hard to parse, with it all being lower same case.

Can we name it something like `macOSlatest` or even `macOS-latest`?

Comment (weshaggard) [reply eng/pipelines/templates/stages/platform-matrix-live.json:8]: @benbp I suspect we could do this with the displayNames property of the configuration correct?

**Comment by weshaggard** [comment]
Looks like the changes made in https://github.com/Azure/azure-sdk-for-cpp/pull/5681 fixed the macos build issues.

File: eng/pipelines/templates/stages/platform-matrix-live.json (hunk 2 above)
Comment (ahsonkhan) [reply eng/pipelines/templates/stages/platform-matrix-live.json:8]: @benbp how can we update the pipeline name to be `MacOS-latest` instead of `macoslatest`?
